internal/resources/policy/type/security/input_schema.go

```diff
@@ -0,0 +1,157 @@
+)
+
+var inputSchema = &schema.Schema{
+	Type:        schema.TypeList,
+	Description: "Input for the security policy",
+	Required:    true,
+	MaxItems:    1,
+	MinItems:    1,
+	ForceNew:    true,
+	Elem: &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			reciperesource.BaselineKey: reciperesource.Baseline,
+			reciperesource.CustomKey:   reciperesource.Custom,
+			reciperesource.StrictKey:   reciperesource.Strict,
+		},
+	},
+}
+
+type (
+	recipe string
+	// InputRecipe is a struct for all types of security policy inputs.
+	inputRecipe struct {
+		recipe        recipe
+		inputBaseline *policyrecipesecuritymodel.VmwareTanzuManageV1alpha1CommonPolicySpecSecurityV1Baseline
+		inputCustom   *policyrecipesecuritymodel.VmwareTanzuManageV1alpha1CommonPolicySpecSecurityV1Custom
+		inputStrict   *policyrecipesecuritymodel.VmwareTanzuManageV1alpha1CommonPolicySpecSecurityV1Strict
+	}
+)
+
+func constructInput(data []interface{}) (inputRecipeData *inputRecipe) {
+	if len(data) == 0 || data[0] == nil {
+		return inputRecipeData
+	}
+
+	inputData, _ := data[0].(map[string]interface{})
+
+	if v, ok := inputData[reciperesource.BaselineKey]; ok {
+		if v1, ok := v.([]interface{}); ok && len(v1) != 0 {
+			inputRecipeData = &inputRecipe{
+				recipe:        baselineRecipe,
+				inputBaseline: reciperesource.ConstructBaseline(v1),
+			}
+		}
+	}
+
+	if v, ok := inputData[reciperesource.CustomKey]; ok {
+		if v1, ok := v.([]interface{}); ok && len(v1) != 0 {
+			inputRecipeData = &inputRecipe{
+				recipe:      customRecipe,
+				inputCustom: reciperesource.ConstructCustom(v1),
+			}
+		}
+	}
+
+	if v, ok := inputData[reciperesource.StrictKey]; ok {
+		if v1, ok := v.([]interface{}); ok && len(v1) != 0 {
+			inputRecipeData = &inputRecipe{
+				recipe:      strictRecipe,
+				inputStrict: reciperesource.ConstructStrict(v1),
+			}
+		}
+	}
+
+	return inputRecipeData
+}
+
+func flattenInput(inputRecipeData *inputRecipe) (data []interface{}) {
+	if inputRecipeData == nil {
+		return data
+	}
+
+	flattenInputData := make(map[string]interface{})
+
+	switch inputRecipeData.recipe {
+	case baselineRecipe:
+		flattenInputData[reciperesource.BaselineKey] = reciperesource.FlattenBaseline(inputRecipeData.inputBaseline)
+	case customRecipe:
+		flattenInputData[reciperesource.CustomKey] = reciperesource.FlattenCustom(inputRecipeData.inputCustom)
+	case strictRecipe:
+		flattenInputData[reciperesource.StrictKey] = reciperesource.FlattenStrict(inputRecipeData.inputStrict)
+	}
+
+	return []interface{}{flattenInputData}
+}
+
+func validateInput(ctx context.Context, diff *schema.ResourceDiff, i interface{}) error {
+	value, ok := diff.GetOk(specKey)
+	if !ok {
+		return fmt.Errorf("spec: %v is not valid: minimum one valid spec block is required", value)
+	}
+
+	data, _ := value.([]interface{})
+
+	if len(data) == 0 || data[0] == nil {
+		return fmt.Errorf("spec data: %v is not valid: minimum one valid spec block is required", data)
```

Comment:
Follow proper logging statements. 
Follow the previous examples

Reply:
This is not returned as a log, instead it is shown as an error (along with the diff) while running terraform plan.